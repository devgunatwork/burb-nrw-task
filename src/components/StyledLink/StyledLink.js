import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styles from './StyledLink.module.scss';

const StyledLink = ({ children, to }) => (
  <Link className={styles.root} to={to}>
    {children}
  </Link>
);

StyledLink.defaultProps = {
  children: '',
};
StyledLink.propTypes = {
  children: PropTypes.node,
  to: PropTypes.string.isRequired,
};

export default StyledLink;
