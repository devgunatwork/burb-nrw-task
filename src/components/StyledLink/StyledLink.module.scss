.root {
  color: inherit;
  text-decoration: none;

  &:hover {
    color: #888888;
  }
}
