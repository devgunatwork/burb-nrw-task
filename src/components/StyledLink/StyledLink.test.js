import React from 'react';
import StyledLink from './StyledLink';

describe('components', () => {
  describe('StyledLink', () => {
    describe('UI', () => {
      it('should match the snapshot', () => {
        const wrapper = shallow(<StyledLink to="/test" />);

        expect(wrapper).toMatchSnapshot();
      });
    });
  });
});
