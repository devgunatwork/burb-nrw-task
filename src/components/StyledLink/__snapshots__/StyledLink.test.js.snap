// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components StyledLink UI should match the snapshot 1`] = `
<Link
  className="root"
  to="/test"
/>
`;
